### DISCUSSION TITLE
Layers workflow

### DISCUSSION THREAD

**Discussion by yolov** [Autodesk/maya-usd#1318] [Q&A]
Hi all,

A lot seems not entirely clear when it comes to the workflow with the USD Layer Editor.
I bump into a few issues all the time and I can't seem to find any useful information out there, so I'll be very happy if anyone can jump on this discussion.

*) Adding sub-layer
I have a simple example, where I have a cube and a layer that overrides its transformation.
Loading my cube as a stage makes the cube the top-level layer.
By loading a sub-layer transform override prompts to save (overwrite) the top layer (i.e. my cube) which is not the desired result for me. I would expect to save a *new* USD file that points to the cube and its override. As adding a parent layer to the top layer is not allowed, the only other option is starting the workflow by first adding an empty layer, then the cube as a sub-layer, then the override as a sub-sub-layer. Is this the intended workflow, or am I using it wrong?

*) Muting layers
Muting layers doesn't really seem to do anything, other than re-read the stage (which results in a UI-freeze) and change the coloring of the layer. Am I wrong to expect that when a layer is muted, the edits done on it should go away (both in viewport and rendering)? If someone can confirm if this is the intended behavior or if it's a bug, I'll be more than happy to report an issue in this project.

Thanks in advance,
Alex Yolov

**Comment by yolov** [comment]
So I've figured this one out, if this helps anyone:
*) My edits need to be on the same level. I.e. I can have a layer that is its own file. My cube and its override should be sub-layers of that other layer and the cube and the override need to be on the same layer level, not parented to each other.
*) Muting works only with relative paths, i.e. usd files loaded in those layer need to be with relative paths for muting to work. Muting layers that load usd files with absolute paths doesn't work. I've added an issue for this, as it seems like a bug.

**Reply by ndeebook** [reply]
I came to the same conclusion on how to organize layers. Here is how I would describe it:
* container_layer.usd
    * edits_layer.usd
    * layer_to_edit.usd

**Reply by yolov** [reply]
Thanks, you've actually put it a lot more clearly :)

**Comment by santosd** [comment]
Hi @yolov,

Thank you for posting the information you were able to find and workflow that works for you. Let us know if you have any further questions. I did see #1340 is logged for Muting Layers issue. I have logged that internally for our team to investigate.

Regards,

David Santos
